File: LeetCode/Java/0643 - Maximum Average Subarray I.java
/*
643. Maximum Average Subarray I
https://leetcode.com/problems/maximum-average-subarray-i/
*/

class Solution {
    public double findMaxAverage(int[] nums, int k) {

        double max = Double.NEGATIVE_INFINITY, avg = 0, sum = 0;

        for (int j = 0; j < nums.length; j++) {
            sum += nums[j];

            if (j >= k - 1) {
                avg = sum / k;
                max = Math.max(avg, max);
                sum -= nums[j + 1 - k];
            }

        }
        return max;
    }
}

File: LeetCode/Python/0643 - Maximum Average Subarray I.py
'''
643. Maximum Average Subarray I
https://leetcode.com/problems/maximum-average-subarray-i/
'''

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

      
        if len(nums) == 1:
            return nums[0]

        
        temp_sum = sum(nums[0:k]) 

        avg = temp_sum / k

        i = 1
        while i <= len(nums)-k:

            temp_sum  =  temp_sum - nums[i-1] + nums[i+k-1]
            temp_avg = temp_sum/k

            avg = max(temp_avg, avg)

            i = i+ 1


        return avg


